rust/src/action/parquet_read/mod.rs

```diff
@@ -153,7 +153,19 @@ impl Add {
                         "minValues" => match record.get_group(i) {
                             Ok(row) => {
                                 for (name, field) in row.get_column_iter() {
-                                    stats.min_values.insert(name.clone(), field.into());
+                                    match field {
+                                        Field::Null => {},
+                                        _ => {
+                                            match field.try_into() {
+                                                Ok(values) => {
+                                                    stats.min_values.insert(name.clone(), values);
+                                                }
+                                                _ => {
+                                                    log::warn!("Unexpected type of minValues field, got: {}", field);
+                                                }
```

Comment:
It's unfortunate that we are duplicating this warning here; it would be nice if we could somehow consolidate this with the one below.

Reply:
I'm afraid I might need a clearer pointer than that 😅. The 2 warnings are coming from different `Err`s so as I understand both need to be handled. I was hoping to find a way to share more of the code between minValues, maxValues and nullCounts but I think that would require understanding Generics in rust.

Reply:
It is quite tricky, having looked at it more. The warning was being generated separately for top-level fields and their children. I created a PR with suggested changes that will simplify some of the logic and avoid the duplication of this warning. https://github.com/Tom-Newton/delta-rs/pull/26

Reply:
Thanks @wjones127. I like to have a go at these things, despite my lacking rust knowledge, but I'm grateful that you are always so helpful.